sources/packages/api/src/services/sfas/sfas-part-time-application.service.ts

```diff
@@ -46,6 +46,51 @@ export class SFASPartTimeApplicationsService
     await this.repo.save(application, { reload: false, transaction: false });
   }
 
+  /**
+   * Validates before an application submission to see if there is an overlapping SFAS part-time application existing.
+   * @param sin
+   * @param birthDate
+   * @param lastName
+   * @param studyStartDate
+   * @param studyEndDate
+   * @returns SFAS part-time application.
+   */
+  async validateDateOverlap(
+    sin: string,
+    birthDate: Date,
+    lastName: string,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<SFASPartTimeApplications> {
+    return this.repo
+      .createQueryBuilder("sfasPTApplication")
+      .select(["sfasPTApplication.id"])
+      .innerJoin("sfasPTApplication.individual", "sfasPTstudent")
+      .where("lower(sfasPTstudent.lastName) = lower(:lastName)", { lastName })
```

Comment:
Why are we searching for an application to match, only with the last name of the student?

Reply:
Lastname + DOB + SIN is the formula to match a student in SFAS individual table.